src/redux/reducers/movies-list_reducer.js

```diff
@@ -0,0 +1,62 @@
+import actionTypes from '../actions/actionTypes.js';
+
+const initialState = {
+  movies: {
+    items: [],
+    totalCount: 0,
+  },
+  fetching: false,
+  searchParams: {
+    filter: [],
+    sortBy: 'release_date',
+    sortOrder: 'asc',
+  },
+};
+
+export default function moviesListReducer(state = initialState, action) {
+  switch (action.type) {
+    case actionTypes.FETCH_MOVIES_START:
+      return {
+        ...state,
+        searchParams: {
+          ...initialState.searchParams,
+          ...action.payload,
+        },
+        fetching: true,
+      };
+    case actionTypes.FETCH_MOVIES_SUCCESS:
+       return {
+         ...state,
+         movies: {
+          items: action.payload.data,
+          totalCount: action.payload.totalAmount,
+         },
+         fetching: false,
+       };
+    case actionTypes.FETCH_MOVIES_FAILURE:
+      return {...state, fetching: false};
```

Comment:
You could also add 'error' field to state for failure cases

Reply:
It makes sense if we'll have a component to display error details